plugin/storage/grpc/proto/storage.proto

```diff
@@ -0,0 +1,188 @@
+        GrpcStoragePluginError error = 2;
+    }
+}
+
+message GetServicesRequest {}
+
+message GetServicesSuccess {
+    repeated string services = 1;
+}
+
+message GetServicesResponse {
+    oneof response {
+        GetServicesSuccess success = 1;
+        GrpcStoragePluginError error = 2;
+    }
+}
+
+message GetOperationsRequest {
+    string service = 1;
+}
+
+message GetOperationsSuccess {
+    repeated string operations = 1;
+}
+
+message GetOperationsResponse {
+    oneof response {
+        GetOperationsSuccess success = 1;
+        GrpcStoragePluginError error = 2;
+    }
+}
+
+message TraceQueryParameters {
+    string service_name = 1;
+    string operation_name = 2;
+    map<string, string> tags = 3;
+    google.protobuf.Timestamp start_time_min = 4 [
+                                                 (gogoproto.stdtime) = true,
+                                                 (gogoproto.nullable) = false
+                                                 ];
+    google.protobuf.Timestamp start_time_max = 5 [
+                                                 (gogoproto.stdtime) = true,
+                                                 (gogoproto.nullable) = false
+                                                 ];
+    google.protobuf.Duration duration_min = 6 [
+                                              (gogoproto.stdduration) = true,
+                                              (gogoproto.nullable) = false
+                                              ];
+    google.protobuf.Duration duration_max = 7 [
+                                              (gogoproto.stdduration) = true,
+                                              (gogoproto.nullable) = false
+                                              ];
+    int32 num_traces = 8;
+}
+
+message FindTracesRequest {
+    TraceQueryParameters query = 1;
+}
+
+message FindTraceIDsRequest {
+    TraceQueryParameters query = 1;
+}
+
+message FindTraceIDsSuccess {
+    repeated bytes trace_id = 1 [
+                                (gogoproto.nullable) = false,
+                                (gogoproto.customtype) = "github.com/jaegertracing/jaeger/model.TraceID",
+                                (gogoproto.customname) = "TraceIDs"
+                                ];
+}
+
+message FindTraceIDsResponse {
+    oneof response {
+        FindTraceIDsSuccess success = 1;
+        GrpcStoragePluginError error = 2;
+    }
+}
+
+message EmptyResponse {
+
+}
+
+message GrpcStoragePluginError {
+    string message = 1;
+}
+
+service GrpcSpanWriterStoragePlugin {
+    // spanstore/Writer
+    rpc WriteSpan(WriteSpanRequest) returns (WriteSpanResponse);
+}
+
+service GrpcSpanReaderStoragePlugin {
+    // spanstore/Reader
+    rpc GetTrace(GetTraceRequest) returns (GetTraceResponse);
+    rpc GetServices(GetServicesRequest) returns (GetServicesResponse);
+    rpc GetOperations(GetOperationsRequest) returns (GetOperationsResponse);
+    rpc FindTraces(FindTracesRequest) returns (stream Trace);
```

Comment:
Returning a stream here solves the issue surfaced in the other PR where grpc message limits were being hit.

Reply:
hm, not sure it completely solves those issues unless we allow multiple Trace entries in the batch for the same trace. A single trace can still easily be larger than the msg limit. 

Since this is a storage API, I would rather go with returning `stream FindTracesResponseChunk` where each chunk contains `[]Span`

Reply:
Out of curiosity what sort of size (and number of spans) would you expect to see the largest of traces reach?

Reply:
we have seen traces in production with 100,000 spans. At average span size of around 500 bytes, that's 50M. Even 10k spans would add up to 5Mb.

I think eventually we'd want to make the chunk size configurable, but I would start with 1000 or even 100 spans.